platform-server-kernel/src/main/java/uno/cod/platform/server/core/service/TaskService.java

```diff
@@ -15,24 +19,43 @@
 @Transactional
 public class TaskService {
     private final TaskRepository repository;
+    private final EndpointRepository endpointRepository;
+    private final OrganizationRepository organizationRepository;
 
     @Autowired
-    public TaskService(TaskRepository repository) {
+    public TaskService(TaskRepository repository, EndpointRepository endpointRepository, OrganizationRepository organizationRepository) {
         this.repository = repository;
+        this.endpointRepository = endpointRepository;
+        this.organizationRepository = organizationRepository;
     }
 
     public void save(TaskCreateDto dto) {
+        Endpoint endpoint = endpointRepository.findOne(dto.getEndpointId());
+        if (endpoint == null) {
+            throw new IllegalArgumentException("endpoint not valid");
+        }
+        Organization organization = null;
```

Comment:
tasks should always relate to a organization, we should set a public flag for this, if a non-coduno organizations owns the tasks, they should be able to update it etc, it's not possible anymore if we "unassign" public tasks from organizations